src/AwesomeCMSCore/Modules/AwesomeCMSCore.Modules.Account/Repositories/GroupRespository.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+using AwesomeCMSCore.Modules.Account.ViewModels;
+using AwesomeCMSCore.Modules.Entities.Data;
+using AwesomeCMSCore.Modules.Entities.Entities;
+using AwesomeCMSCore.Modules.Repositories;
+using Microsoft.EntityFrameworkCore;
+
+namespace AwesomeCMSCore.Modules.Account.Repositories
+{
+    public class GroupRespository : IGroupRespository
+    {
+        private readonly ApplicationDbContext _context;
+        public GroupRespository(ApplicationDbContext _dbContext)
+        {
+            _context = _dbContext;
+        }
+        public Task Create(GroupViewModel groupViewModel)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<IEnumerable<GroupViewModel>> GroupList()
+        {
+            var groupList = await (from applicationgroup in _context.ApplicationGroups
```

Comment:
Can you use unit of work like this
https://github.com/Awesome-CMS-Core/Awesome-CMS-Core/blob/master/src/AwesomeCMSCore/Modules/AwesomeCMSCore.Modules.Admin/Services/TagService.cs#L34

Reply:
you mean SingleOrDefaultAsync() ?

Reply:
No I mean you can query like this
  var tagData = await _unitOfWork.Repository<Tag>().FindBy(x => x.UserId == _currentUserId)

Because UnitOfWork is wrap on top ApplicationDBContext and we need to avoid direct call to ApplicationDbContext

Reply:
Then mapper?

Reply:
we need mapper to return only properties we need . Best practice is not return and entity but return view model

Reply:
okay